adminapi/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -291,6 +293,74 @@ public void deleteUser(@Nonnull String accessKey)
             null)) {}
   }
 
+  /**
+   * set bucket quota size
+   * @param bucketName bucketName
+   * @param size the capacity of the bucket
+   * @param unit the quota unit of the size argument
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws IOException thrown to indicate I/O error on MinIO REST operation.
+   */
+  public void setBucketQuota(@Nonnull String bucketName, long size, @Nonnull QuotaUnit unit)
+          throws IOException, NoSuchAlgorithmException, InvalidKeyException {
+    Map<String, Object> quotaEntity = new HashMap<>(4);
```

Comment:
Why `4`?

Reply:
Because an empty HashMap with the default initial capacity (16), this reduces array capacity creation.

Reply:
If optimization matters, just create JSON string directly like
```java
String s = "{\"quota\":" + unit.toBytes(size);
if (size > 0) s += ",\"quotatype\":\"hard\";
s += "}";
```

Reply:
This optimization is not very important here. So I remove 4